### ISSUE TITLE
Feature request: HSTS support

### ISSUE THREAD

**Issue by nertzy** [cloudfoundry/staticfile-buildpack#82]
In https://github.com/cloudfoundry/staticfile-buildpack/pull/37 there was a discussion of how to best implement [HTTP Strict Transport Security](https://developer.mozilla.org/en-US/docs/Web/Security/HTTP_strict_transport_security).

I agree with that conversation that the best place to store this configuration is the Staticfile.

One caveat is that I wouldn't want this activated in any environments where HTTPS wasn't enabled, since any users that hit the site while it's deployed that way would be blocked from hitting the HTTP site for a long time.

Therefore it probably makes the most sense to require the `FORCE_HTTPS` environment variable to be set as well before this would take any effect. (Maybe that should be moved to the Staticfile as well?)

**Comment by flavorjones** [comment]
As I mentioned in the previous ticket, I'm not convinced that environment variables are the appropriate configuration mechanism.

**Comment by flavorjones** [comment]
I'd prefer for this to be configurable through the `Staticfile`

**Comment by nertzy** [comment]
In my opinion the ideal situation for all settings is that you can set a default value in the Staticfile but that you could also override each of those settings with environment variables. 

I could see someone wanting HSTS in public-facing production-like environments, but not in lower environments if they don't want the overhead of setting up SSL everywhere.

**Comment by nertzy** [comment]
Depends on whether we consider each thing to be "internal application config" or not as per https://12factor.net/config.

**Comment by jchesterpivotal** [comment]
My own opinion is that HSTS should be a switch in the `Staticfile`, not an environment variable. These 

Environment variables are, ideally, only there to inject app behaviour. This buildpack blurs the line between app and buildpack, but I think we should maintain the distinction nonetheless. We broke this rule for `FORCE_HTTPS`, which we ought to deprecate, but in general we've remained consistent.

In particular, Staticfile *does not build live applications*. By extension from "environment variables are for apps", if there are no apps, there should be no environment variables.

Staticfile's particular brand of snowflakery is that there's no accepted ecosystem tooling and idiom that we're relying on to create and enforce conventions. For example, in Ruby, we don't "think" about these questions much -- we just delegate to Bundler and rubygems and let them set the policy for us.

While having no default idiom or packaging seems like freedom, it's also a scope magnet. I'm personally wary of adding multiple layers of configurability because we will have to *entirely* own the upkeep.
